minishift_sanity.py

```diff
@@ -92,90 +92,164 @@ def test_ms_start(self):
         else:
             self.fail("Minishift start failed")
             provision_path = None
+    """
     
-    """        
+    def test_ms_setup_cdk(self):
+        '''
+        Testing if setup-cdk command is able to setup CDK properly:
+        RHEL iso and OCP binary are in place
+        '''
+        cmd = "minishift setup-cdk"
+        self.log.info("Running command: " + cmd)
+        child = pexpect.spawn(cmd)
+        index = child.expect(["CDK 3 setup complete.", pexpect.EOF, pexpect.TIMEOUT], timeout=30)
+        if not index==0:
+            self.fail("CDK setup failed")
+        self.log.info("Checking if rhel.iso exists")
+        exists = os.path.isfile( os.environ['HOME'] + "/.minishift/cache/iso/minishift-rhel.iso" )
+        if not exists:
+            self.fail("minishift-rhel.iso is not present")
+        self.log.info("Checking if oc binary exists")
+        exists = os.path.isfile( os.environ['HOME'] + "/.minishift/cache/oc/v3.4.0/oc" )
+        if not exists:
+            self.fail("oc binary is not present")
+        self.log.info("Checking if config.json exists")
+        exists = os.path.isfile( os.environ['HOME'] + "/.minishift/config/config.json" )
+        if not exists:
+            self.fail("config.json binary is not present")
+        self.log.info("CDK Installation successful")
+
     def test_ms_start(self):
-        if iso_url == "":
-            cmd = "minishift start --vm-driver " +self.Hypervisor_Provider
-        else:
-            cmd = "minishift start --vm-driver " +self.Hypervisor_Provider +" --iso-url " +self.iso_url
```

Comment:
minishift with rhel iso requires RHN credentials so the command would be look like this  

cmd = "minishift start --vm-driver " +self.Hypervisor_Provider +" --iso-url " +self.iso_url "--username " +self.self.params.get('RHN_USERNAME') +"  --password  " +self.params.get('RHN_PASSWORD')

NOTE :  RHN_USERNAME and RHN_PASSWORD are set in config.yaml

IMO its beter to have one more "else if" condition. Feel free to comment